detectable/src/main/java/com/synopsys/integration/detectable/detectables/dart/pubdep/DartPubDepDetectable.java

```diff
@@ -0,0 +1,103 @@
+import com.synopsys.integration.detectable.detectable.result.FilesNotFoundDetectableResult;
+import com.synopsys.integration.detectable.detectable.result.PassedDetectableResult;
+import com.synopsys.integration.detectable.detectable.result.PubSpecLockNotFoundDetectableResult;
+import com.synopsys.integration.detectable.extraction.Extraction;
+import com.synopsys.integration.detectable.extraction.ExtractionEnvironment;
+
+@DetectableInfo(language = "Dart", forge = "Dart", requirementsMarkdown = "Files: pubspec.yaml, pubspec.lock.<br/><br/>Executable: dart, flutter")
+public class DartPubDepDetectable extends Detectable {
+    public static final String PUBSPEC_YAML_FILENAME = "pubspec.yaml";
+    public static final String PUBSPEC_LOCK_FILENAME = "pubspec.lock";
+
+    private final FileFinder fileFinder;
+    private DartResolver dartResolver;
+    private FlutterResolver flutterResolver;
+    private PubDepsExtractor pubDepsExtractor;
+    private DartPubDepsDetectableOptions dartPubDepsDetectableOptions;
+
+    private File pubspecYaml;
+    private File pubspecLock;
+
+    private ExecutableTarget dartExe;
+    private ExecutableTarget flutterExe;
+
+    public DartPubDepDetectable(DetectableEnvironment environment, FileFinder fileFinder, PubDepsExtractor pubDepsExtractor, DartPubDepsDetectableOptions dartPubDepsDetectableOptions, DartResolver dartResolver,
+        FlutterResolver flutterResolver) {
+        super(environment);
+        this.fileFinder = fileFinder;
+        this.pubDepsExtractor = pubDepsExtractor;
+        this.dartPubDepsDetectableOptions = dartPubDepsDetectableOptions;
+        this.dartResolver = dartResolver;
+        this.flutterResolver = flutterResolver;
+
+    }
+
+    @Override
+    public DetectableResult applicable() {
+        PassedResultBuilder passedResultBuilder = new PassedResultBuilder();
+
+        pubspecYaml = fileFinder.findFile(environment.getDirectory(), PUBSPEC_YAML_FILENAME);
+        pubspecLock = fileFinder.findFile(environment.getDirectory(), PUBSPEC_LOCK_FILENAME);
+
+        if (pubspecLock != null) {
+            passedResultBuilder.foundFile(pubspecLock);
+        }
+        if (pubspecYaml != null) {
+            passedResultBuilder.foundFile(pubspecYaml);
+        }
+
+        if (pubspecLock != null && pubspecYaml == null) {
+            return new FileNotFoundDetectableResult(PUBSPEC_YAML_FILENAME);
+        }
+
+        if (pubspecYaml == null && pubspecLock == null) {
+            return new FilesNotFoundDetectableResult(PUBSPEC_LOCK_FILENAME, PUBSPEC_YAML_FILENAME);
+        } else {
+            return passedResultBuilder.build();
+        }
+
+    }
+
+    @Override
+    public DetectableResult extractable() throws DetectableException {
+        if (pubspecLock == null && pubspecYaml != null) {
+            return new PubSpecLockNotFoundDetectableResult(environment.getDirectory().getAbsolutePath());
```

Comment:
Why is this file check done in `extractable()` when the combination of both missing and just the yaml are done in `applicable()`